### DISCUSSION TITLE
PANDA motif filtering

### DISCUSSION THREAD

**Discussion by tithuytrang** [netZoo/netZooR#263] [Q&A] [answered]
Dear NetZooR team,
I'm Trang Truong, a PhD student from Deakin University. Many thanks for the great package!
While using PandaR for regulatory network construction, several TFs in the regNet output didn't exist (not expressed or already filtered due to low reads) in the input expression. I tried "remove.missing.motif = T" for "legacy" mode but it's not fixed. It seemed your source code (below) only filtered missing targets but not the missing TFs themselves. I would modify to `motif <- motif[which(motif[, 2] %in% rownames(expr) & motif[, 1] %in% rownames(expr)), ]`.
I assume both TFs and targets should be expressed sufficiently to be included in the network but could there be a justification for not filtering for missing TFs that I'm not aware of?
```  
if (remove.missing.motif) {
        n <- nrow(motif)
        motif <- motif[which(motif[, 2] %in% rownames(expr)), ]
        message(sprintf("%s motif edges removed that targeted genes missing in expression data", 
          n - nrow(motif))) }
```
Thanks,
Trang

**Comment by marouenbg** [comment, marked as answer]
Hi @tithuytrang ,
If I understood correctly, panda includes TFs in the final network, that were not originally included in the expression matrix.
In the final network, PANDA actually includes TFs from motif data as well using 3 different modes:
- union: TFs from the union of TFs in expression and motif
- intersection: TFs in the intersection of expression and motif
- legacy: TFs from motif only
Actually, PANDA does not use the expression matrix for TF assessment, it uses it rather for gene to gene similarity computation.

I hope that helps,
Marouen

**Reply by tithuytrang** [reply]
Thanks Marouen! I understood PANDA results can include TFs not existing in the expression matrix based on the running mode. However, "intersection" mode also only filtered missing targets but not TFs and we had same issue with missing TFs showing in final results. I wonder why after all filtering parameters in PANDA (PPIs, motifs, genes), TFs are still kept in the final matrix even if they don't (or barely) express?

I checked the play data from data(pandaToyData) and this didn't happen, probably both TFs and targets of the motif matrix have been filtered based on genes in expression matrix. There's no mentioning for this step so I'm not sure if we have to preprocess manually before inputting to PANDA?

Thanks,
Trang

**Reply by marouenbg** [reply]
Hi @tithuytrang ,
PANDA does not  use gene expression to filter TFs, it does so only for genes, therefore it is common to have TFs in the final network that are not expressed in the gene expression matrix.
As you said, you can filter the TFs in the motif matrix based on gene expression before running PANDA.

**Reply by tithuytrang** [reply]
Thanks very much @marouenbg for explaining!

**Reply by marouenbg** [reply]
We do this because most TFs are lowly expressed and therefore their activity doesn't seem to always correlate with their RNA abundance, but in practice we filter TFs using gene expression like you're suggesting.

**Reply by tithuytrang** [reply]
That makes sense @marouenbg. I have a question for the GRAND database but there's no Discussion in its github, could you advise me where to post it?

**Reply by marouenbg** [reply]
Sure, you can use issues here https://github.com/QuackenbushLab/grand
